### ISSUE TITLE
[BUG] (hidden) input id is reported as invalid

### ISSUE THREAD

**Issue by puschie286** [IBMa/equal-access#568]
## What tool(s) are you using?

IBM Accessibility
Equal Access Toolkit:
Accessibility Checker
Version 3.1.12

- `accessibility-checker-engine`

## Detailed description

Some labels are reported as "The value "menu0" of the 'for' attribute is not the 'id' of a valid <input> element" but the id is correct and valid. These elements are visual / functional only, they have aria-hidden attribute but still reported as issues

https://codesandbox.io/embed/issue-568-forked-h5mzr

**Comment by tombrunet** [comment]
https://codesandbox.io/s/issue-568-j4ljn - I am unable to reproduce the error with the code that you provided. Two comments:
1) In is invalid to mark a focusable control as `aria-hidden`, therefore the `aria-hidden` on the `input` is a violation.
2) You may have some other element on your page that has an id of `menu0` that isn't a labellable element.

**Comment by tombrunet** [comment]
Thank you, that helps. So it's a visible label pointing to an invisible checkbox (CSS display:none). I will raise this with the team for further discussion. It feels problematic since the label isn't in the tab order by default, so you have something that's reachable by mouse but not keyboard. That might be fixed with a tabindex, but then the label may not reflect the checked state of the checkbox. It's an odd pattern, and I believe the rule may be correct. It will help for discussion if we better understand what you're trying to do with that label / checkbox combination. Thank you.

**Comment by puschie286** [comment]
the purpose is to provide a dynamic animated menu without javascript - i know that you can achiev this with the checkbox alone but we heavly using checkboxes for other stuff and we had problems to isolate the css between these parts, so this was the easiest solution for that time.
beside the implementation flaw - its just a visual thing for mouse users, the actual links are all reachable by keyboard ( no problems with the menu ). so we "just" need to hide this construct from non-mouse users ( and its the last error reported by your tool, so we would like to make sure that doesnt disturbed some users )

**Comment by tombrunet** [comment]
Can you give more information on how a keyboard user would interact with this menu? It almost seems like you're trying to create a separate experience, which usually makes a poor assumption that accessibility needs are separated into distinct buckets, which isn't a good assumption.

**Comment by puschie286** [comment]
we delivered our product with this structure to one of our customer and it looks like they dont have any problems to use it with keyboard - so it can be ignored if anyone has a similar problem
